### DISCUSSION TITLE
How do I clear all the data from my github page?

### DISCUSSION THREAD

**Discussion by bythckr** [community/community#23325] [New to GitHub]
<p>I have done hobby projects &amp; forked some interesting projects to tinker around in my official github page. Now I was clear it all. Is there a one button to do it or do I have to delete each repo (100+) &amp; unstar (200+) each projects?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>You could use the <a href="https://github.com/octokit" rel="nofollow noopener">Octokit</a> to create a script that pulls all of your repositories and then use the <a href="https://developer.github.com/v3/repos/#delete-a-repository" rel="nofollow noopener">delete endpoint</a> to delete all the content. Similarly, you can manage your starred repositories by listing all of your <a href="https://developer.github.com/v3/activity/starring/#list-repositories-being-starred" rel="nofollow noopener">starred repositories</a> then write your script to select repos you want to <a href="https://developer.github.com/v3/activity/starring/#unstar-a-repository" rel="nofollow noopener">unstar</a>.<br>
Of course, there is no harm in keeping the stars as it helps GitHub recommend similar projects in the discovery view of your news feed. I hope this helps!</p>

**Comment by bythckr** [comment]
<p>I am not sure how to script with octokit, I have not written scripts. Do you have some sample code in python that I can use to understand? Since your official library in there in node.js, even that is fine.</p>

**Comment by lakshay313** [comment]
<p><a href="https://github.com/lakshay313/github-unstar-repo" rel="nofollow noopener">https://github.com/lakshay313/github-unstar-repo</a></p>
<p>Ruby script to remove starred repos from your profile</p>
